contracts/PoolManager.sol

```diff
@@ -171,12 +171,12 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
     }
 
     /// @inheritdoc IPoolManager
-    function lock(bytes calldata data) external override returns (bytes memory result) {
+    function lock(address lockTarget, bytes calldata data) external override returns (bytes memory result) {
```

Comment:
I imagine we won't see the true gas changes this involves until the tests start calling PoolManager directly....I'd honestly expect gas to go down rather than up? I just dont' see a huge benefit tho, bc ultimately a router will need to call swap, and msg.sender will still be a router..... am I missing something?

Reply:
Yeah gas goes down about 2k since it avoids the extra call!

Basic idea is if a hook needs to know the _swapper_ address that currently requires a trusted router to pass through the swapper data. With this, the swapper could just call the poolmanager directly and hook could use PM's msg.sender to determine the swapper

Reply:
>hook could use PM's msg.sender to determine the swapper

need a way to pass PM's msg.sender into hook. I've discussed one way here: https://www.notion.so/uniswaplabs/V4-Arbitrary-Data-msg-sender-73d4a2d3f43241049249af6278abd2d5?pvs=4#0327195ab2ca4827829e9d64c398a38b

Reply:
>With this, the swapper could just call the poolmanager directly and hook could use PM's msg.sender to determine the swapper

my confusion here is: the hook doesn't get called until we call swap (which is done by the router, not the EOA which called `lock`, so now that msg.sender is lost by the tiem we're in swap.)

I really didn't (don't) like the idea of tracking `msg.sender` in core, but now I'm realizing that it's so hard to do it from the router, bc a hook would have to be 4ever paired to a router which doesn't seem ideal. I think the t-storing the lock's `msg.sender`in core is not going to be straightforward, bc just like locker addr, it will have to be recursive, and I'm still wondering whether we _really_ want to support the permissioned usecase enough to merit the complexity...

Reply:
> my confusion here is: the hook get's called inside swap, which is called by the router, NOT the EOA which called lock, so by the time we're calling the hook, the original msg.sender is lost.

Yes this is true. I addressed this issue above in https://github.com/Uniswap/v4-core/pull/300#discussion_r1256015791.

> I think the t-storing the lock's msg.sender in core is not going to be straightforward, bc just like locker addr, it will have to be recursive
I don't follow - can you be more specific? I believe the method I've described in https://github.com/Uniswap/v4-core/pull/300#discussion_r1256015791 is sufficient.